src/hist/core.py

```diff
@@ -2,29 +2,270 @@
 import matplotlib
 import matplotlib.pyplot as plt
 import matplotlib.patches as patches
+from matplotlib import transforms
 from scipy.optimize import curve_fit
 from uncertainties import correlated_values, unumpy
 from boost_histogram import Histogram
-from typing import Callable, Optional, Tuple
+from typing import Callable, Optional, Tuple, Union
 
 
 class BaseHist(Histogram):
     def __init__(self, *args, **kwargs):
         """
-            Initialize Hist object. Axis params must contain the names.
+            Initialize BaseHist object. Axis params can contain the names.
         """
 
         super().__init__(*args, **kwargs)
         self.names: dict = dict()
         for ax in self.axes:
             if ax.name in self.names:
                 raise Exception(
-                    "Hist instance cannot contain axes with duplicated names."
+                    f"{self.__class__.__name__} instance cannot contain axes with duplicated names."
                 )
             else:
                 self.names[ax.name] = True
 
-    def pull_plot(
+    def project(self, *args: Union[int, str]):
```

Comment:
This returns a Union of the histogram type and the contents type. I can show you how to set that up later.

Reply:
okay.